ambry-router/src/test/java/com.github.ambry.router/InMemoryRouter.java

```diff
@@ -310,30 +327,40 @@ protected static void completeOperation(FutureResult futureResult, Callback call
   private final ConcurrentHashMap<String, InMemoryRouter.InMemoryBlob> blobs;
   private final NotificationSystem notificationSystem;
   private final ClusterMap clusterMap;
+  private final boolean useBlobIdV1;
 
   /**
    * Create a new instance.
    * @param postData the data that came with the POST request as {@link PostData}.
    * @param blobs the list of blobs in memory.
    * @param notificationSystem the notification system to use to notify creation/deletion of blobs.
    * @param clusterMap the cluster map for the cluster.
+   * @param useBlobIdV1 {@code true} to use V1 blob IDs, otherwise use the latest version.
    */
   public InMemoryBlobPoster(PostData postData, ConcurrentHashMap<String, InMemoryRouter.InMemoryBlob> blobs,
-      NotificationSystem notificationSystem, ClusterMap clusterMap) {
+      NotificationSystem notificationSystem, ClusterMap clusterMap, boolean useBlobIdV1) {
```

Comment:
minor: may you can make this generic to take in BlobId version. We just use V1 or default for now. But in future, might be handy

Reply:
done